ui/components/app/confirm-gas-display/confirm-gas-display.test.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { screen, waitFor } from '@testing-library/react';
+
+import { GasEstimateTypes } from '../../../../shared/constants/gas';
+import mockEstimates from '../../../../test/data/mock-estimates.json';
+import mockState from '../../../../test/data/mock-state.json';
+import { renderWithProvider } from '../../../../test/jest';
+import configureStore from '../../../store/store';
+
+import { GasFeeContextProvider } from '../../../contexts/gasFee';
+import ConfirmGasDisplay from './confirm-gas-display';
+
+jest.mock('../../../store/actions', () => ({
+  disconnectGasFeeEstimatePoller: jest.fn(),
+  getGasFeeEstimatesAndStartPolling: jest
+    .fn()
+    .mockImplementation(() => Promise.resolve()),
+  addPollingTokenToAppState: jest.fn(),
+  getGasFeeTimeEstimate: jest.fn().mockImplementation(() => Promise.resolve()),
+}));
+
+const render = ({ transactionProp = {}, contextProps = {} } = {}) => {
+  const store = configureStore({
+    ...mockState,
+    ...contextProps,
+    metamask: {
+      ...mockState.metamask,
+      accounts: {
+        [mockState.metamask.selectedAddress]: {
+          address: mockState.metamask.selectedAddress,
+          balance: '0x1F4',
+        },
+      },
+      preferences: {
+        useNativeCurrencyAsPrimaryCurrency: true,
+      },
+      gasFeeEstimates: mockEstimates[GasEstimateTypes.feeMarket],
+    },
+  });
+
+  return renderWithProvider(
+    <GasFeeContextProvider transaction={transactionProp}>
+      <ConfirmGasDisplay />
+    </GasFeeContextProvider>,
+    store,
+  );
+};
+
+describe('ConfirmGasDisplay', () => {
+  it('should match snapshot', async () => {
+    const { container } = render();
+    await waitFor(() => {
+      expect(container).toMatchSnapshot();
+    });
+  });
+  it('should render gas display labels for EIP1559 transcations', async () => {
+    render({
+      transactionProp: {
+        txParams: {
+          gas: '0x5208',
+          maxFeePerGas: '0x59682f10',
+          maxPriorityFeePerGas: '0x59682f00',
+        },
+        userFeeLevel: 'medium',
+      },
+    });
+    await waitFor(() => {
```

Comment:
seems the test are passing if we remove `async` and `await waitFor(() => {` in this test and the one below